File: python-version/LC3195/minimum_area.py
from typing import List


class Solution:
    @staticmethod
    def minimumArea(grid: List[List[int]]) -> int:
        if not grid or not grid[0]:
            return 0
        m, n = len(grid), len(grid[0])
        min_row, min_col = m, n
        max_row, max_col = 0, 0
        for i in range(m):
            for j in range(n):
                if grid[i][j] == 1:
                    min_row = min(min_row, i)
                    min_col = min(min_col, j)
                    max_row = max(max_row, i)
                    max_col = max(max_col, j)
        return (max_row - min_row + 1) * (max_col - min_col + 1)

File: go-version/leetcode/LC3195/minimum_area.go
package LC3195

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/8/22 下午8:23
 * @FilePath: minimum_area
 * @Description: 直接遍历
 */

func minimumArea(grid [][]int) int {
	if len(grid) == 0 || len(grid[0]) == 0 {
		return 0
	}
	m, n := len(grid), len(grid[0])
	minRow, minCol := m, n
	maxRow, maxCol := 0, 0
	for i := 0; i < m; i++ {
		for j := 0; j < n; j++ {
			if grid[i][j] == 1 {
				minRow = min(minRow, i)
				minCol = min(minCol, j)
				maxRow = max(maxRow, i)
				maxCol = max(maxCol, j)
			}
		}
	}
	return (maxRow - minRow + 1) * (maxCol - minCol + 1)
}
